fas2ipa/users.py

```diff
@@ -354,7 +356,14 @@ def migrate_user(self, person):
                     if user_args["full_name"] == "<first-name-unset> <last-name-unset>":
                         del user_args["full_name"]
 
-                    # Avoid overwriting already set fields
+                    # Avoid resetting already set fields
+                    if user_args["faslocale"] is None or user_args["fastimezone"] is None:
+                        ipa_user = self.user_show(username)
+                        if not user_args["faslocale"]:
+                            user_args["faslocale"] = ipa_user.get("faslocale") or "en_US"
+                        if not user_args["fastimezone"]:
+                            user_args["fastimezone"] = ipa_user.get("fastimezone") or "UTC"
```

Comment:
I don't understand the reason for this block: if the user has already been imported (say, in the Fedora run) and we get a user with the same username in a second import (say, in CentOS) that did not set their locales & timezones, then these values will be `None` and they will be removed by the dict loop right below, no?

Reply:
This reflects that users could have been created in IPA with an unset locale, time zone in previous runs (e.g. staging): When modifying an existing IPA user, only apply the defaults to these fields if they're unset in the IPA user, too, not just the processed FAS user.

Reply:
Ah, I see. Indeed in production all users will already exist. I'm a bit afraid that adding the `user_show` call for all existing user (thus all users) will make the import much slower.

Maybe we can run a post-import script to get all users with unset locale & timezone and set the defaults?

Reply:
The `user_show()` call will only be run for the subset of (FAS) users that have an unset locale or time zone (line 360), this affects ~8% of users.

Reply:
Right, it should be fine :-)